File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        int i = 0; //indexes
        for (auto x: nums) {
            if (mp.find(target-x) != mp.end()) return {i,mp.at(target-x)};
            mp.insert({x,i});
            i++;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        tbl = {}

        for i,n in enumerate(nums):
            if tbl.get(target-n, None) != None:
                #means we got the answer
                return [tbl.get(target-n),i]
            else:
                tbl[n] = i
 
        return [0,0] 